backend/dashboard/views/components/PaymentMethods.vue

```diff
@@ -0,0 +1,177 @@
+<template lang='pug'>
+.c-payment-methods-container
+  .c-entry(
+    v-for='item in ephemeral.paymentMethods'
+    :key='item.id'
+  )
+    .inputgroup.c-method-group
+      dropdown.c-method-dropdown(
+        :defaultText='L("Choose...")'
+        :options='config.methodOptions'
+        :isOverlayStyle='true'
+        @select='$event => onDropdownSelect(item.id, $event)'
+      )
+      input.input.no-label.c-method-input(type='text' v-model='item.detail')
+
+    button.is-icon.c-delete-entry-btn(
+      v-if='ephemeral.paymentMethods.length > 1'
+      @click='deleteEntry(item.id)'
+    )
+      i.icon-minus-circle
+
+  .c-add-more
+    button.is-unstyled.c-add-more-btn(v-if='showAddMore' type='button' @click='addEntry')
+      i.icon-plus-circle
+      i18n.is-inline-block Add more
+</template>
+
+<script>
+import { cloneDeep } from '@common/cdLodash.js'
+import Dropdown from '@forms/Dropdown.vue'
+
+const genRandomId = () => Math.random().toString(20).slice(2)
+
+export default {
+  name: 'PaymentMethods',
+  props: {
+    methods: {
+      type: Array,
+      required: true
+    }
+  },
+  components: {
+    Dropdown
+  },
+  computed: {
+    showAddMore () {
+      return this.ephemeral.paymentMethods.length < this.config.methodOptions.length
+    }
+  },
+  data () {
+    return {
+      config: {
+        methodOptions: [
+          { id: 'paypal', name: 'Paypal' },
+          { id: 'bitcoin', name: 'Bitcoin' },
+          { id: 'venmo', name: 'Venmo' }
+        ]
+      },
+      ephemeral: {
+        paymentMethods: [
+          { id: genRandomId(), method: null, detail: '' }
+        ]
+      },
+      syncDebounceId: null
+    }
+  },
+  methods: {
+    onDropdownSelect (targetId, selectedItem) {
+      const found = this.ephemeral.paymentMethods.find(x => x.id === targetId)
+      found.method = selectedItem
+
+      this.$emit('update:methods', cloneDeep(this.ephemeral.paymentMethods))
+    },
+    addEntry () {
+      this.ephemeral.paymentMethods.push({ id: genRandomId(), method: null, detail: '' })
+    },
+    deleteEntry (targetId) {
+      const index = this.ephemeral.paymentMethods.findIndex(x => x.id === targetId)
+      this.ephemeral.paymentMethods.splice(index, 1)
+    },
+    debouncedMethodsSync () {
+      if (this.syncDebounceId) {
+        clearTimeout(this.syncDebounceId)
+      }
+
+      this.syncDebounceId = setTimeout(() => {
+        this.$emit('update:methods', cloneDeep(this.ephemeral.paymentMethods))
+        this.$emit('change')
+      }, 800)
+    }
+  },
+  updated () {
+    this.debouncedMethodsSync()
+  }
```

Comment:
@Silver-IT Setting `this.syncDebounceId` here actually doesn't lead to a component re-render, because it deosn't participate in DOM update, so no infinite loop happening. using `updated()` hook here was the most concise way of keeping things in sync with the parent component in response to the multiple required user-actions (typing to input, dropdown selection, a new entry additon/deletion)

Reply:
Yeah, you are right. I tested it on my side by logging to check how many times the function is called.

According to the document, I can see that
```
'updated' hook is called after the component has updated its DOM tree due to a reactive state change.
```